track_info.c

```diff
@@ -46,8 +46,12 @@ static void track_info_free(struct track_info *ti)
 
 struct track_info *track_info_new(const char *filename)
 {
+	static uint64_t cur_uid = 0;
+	cur_uid++;
```

Comment:
That's not race-free. The one point where it's used is in the seek request to ensure that the track hasn't changed in the meantime.